### ISSUE TITLE
CS8602 false positive

### ISSUE THREAD

**Issue by KeterSCP** [dotnet/roslyn#67062]
**Version Used**: 
4.4.0-6.22559.4 (d7e8a398)

**Steps to Reproduce**:

1. Consider following code snippet:
```c#
using System;
using System.Collections.Generic;

#nullable enable

void Test(List<int>? l1, List<int>? l2)
{
    if (l1 is null && l2 is null) return;

    if (l1?.Count != l2?.Count) return;

    Console.WriteLine(l1.Count);
    Console.WriteLine(l2.Count);
}
```

[SharpLab link](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0ATEBqAPgAQCYBGAWACh8AGAAn2JQG4KKBiAOwFcAbbgQ2DcYNGOwFCWlFDQAqMAM4AXABQAZAJZKAPOvaKAfAH4a3Ymhobtug8e6EAlBQDeFGm5rqAZjWWmP8mi5eGgAyEJNCf0Cebns6AHZmcld3Lx9TQwA6AGEITj0aAEIAXgis3PzFOPxEyXc6YgBOX2IcvL17JPr6Zrs2ys6KAF8gA)

**Diagnostic Id**:

CS8602

**Expected Behavior**:
There should be no CS8602 warnings when accessing `l1` and `l2` properties after checking both if-s

**Actual Behavior**:
There are two CS8602 warnings when accessing `l1` and `l2` properties

**Comment by jcouv** [comment]
Unfortunately, this limitation is by design. We don't track correlated variable states. So after the first `if`, both variables may be `null`. That first `if` has no effect on the nullable state that is tracked. This is a good scenario for using a suppression with `!`.
